packages/common/src/store/pages/chat/sagas.ts

```diff
@@ -155,17 +159,54 @@ function* doCreateChat(action: ReturnType<typeof createChat>) {
   try {
     const audiusSdk = yield* getContext('audiusSdk')
     const sdk = yield* call(audiusSdk)
-    const user = yield* select(getAccountUser)
-    if (!user) {
+    const currentUserId = yield* select(getUserId)
+    if (!currentUserId) {
       throw new Error('User not found')
     }
-    const res = yield* call([sdk.chats, sdk.chats.create], {
-      userId: encodeHashId(user.user_id),
-      invitedUserIds: userIds.map((id) => encodeHashId(id))
-    })
-    const chatId = (res as ChatCreateRPC).params.chat_id
-    const { data: chat } = yield* call([sdk.chats, sdk.chats.get], { chatId })
-    yield* put(createChatSucceeded({ chat }))
+    // Try to get existing chat:
+    const chatId = [currentUserId, ...userIds]
+      .map((id) => encodeHashId(id))
+      .sort()
+      .join(':')
+    try {
+      yield* call(fetchChatIfNecessary, { chatId })
+    } catch {}
+    const existingChat = yield* select((state) => getChat(state, chatId))
+    if (existingChat) {
+      // Simply navigate to the existing chat
+      yield* put(setVisibility({ modal: 'CreateChat', visible: false }))
+      yield* put(goToChat({ chatId: existingChat.chat_id }))
+    } else {
+      // Create new chat and navigate to it
+      yield* call([sdk.chats, sdk.chats.create], {
+        userId: encodeHashId(currentUserId),
+        invitedUserIds: userIds.map((id) => encodeHashId(id))
+      })
+      let chat = null
+      let i = 0
+      // Poll for new chat
+      while (i++ < 10 && !chat) {
```

Comment:
pull out 10 to a constant?

Reply:
oh wait wut.. what is meant by "mutation will wait now"?

Reply:
@stereosteve  PR https://github.com/AudiusProject/audius-protocol/pull/4792 !